MetadataExtractor/Formats/Bmp/BmpReader.cs

```diff
@@ -347,8 +347,13 @@ private static void ReadBitmapHeader(SequentialReader reader, BmpHeaderDirectory
                         }
                         else
                         {
-                            var iccReader = new ByteArrayReader(reader.GetBytes(profileSize));
-                            var iccDirectory = new IccReader().Extract(iccReader);
+                            using var iccBuffer = profileSize <= 256
+                                ? new BufferScope(stackalloc byte[profileSize])
+                                : new BufferScope(profileSize);
```

Comment:
Running this over the test suite shows a `StackOverflowException` here. Were you able to get the regression suite working locally? Let me know if you need a hand with that.

The issue is that `profileSize` is negative.

Perhaps something like:

```suggestion
                            using var iccBuffer = (uint)profileSize <= 256
                                ? new BufferScope(stackalloc byte[profileSize])
                                : new BufferScope(profileSize);
```

That's a clever and performant way to validate the value is also greater than zero, but it's a less common pattern and doesn't read very naturally.

Wish there was a better way to wrap this pattern up, but I can't see it, at least not for all target frameworks we have. For .NET 8 I think we could use inline arrays with something like this:

```c#
internal readonly ref struct BufferScope
{
    private const int StackBufferSize = 256;

    private readonly Buffer _buffer;
    private readonly byte[]? _array;
    private readonly Span<byte> _span;

    public BufferScope(int size)
    {
        if (size < 0)
        {
            throw new ArgumentOutOfRangeException(nameof(size), size, "Cannot be negative.");
        }
        else if (size < StackBufferSize)
        {
            _span = MemoryMarshal.CreateSpan(ref Unsafe.As<Buffer, byte>(ref _buffer), size);
        }
        else
        {
            _array = ArrayPool<byte>.Shared.Rent(size);
            _span = _array.AsSpan(0, size);
        }
    }

    public readonly Span<byte> Span => _span;

    public readonly void Dispose()
    {
        if (_array is null) return;

        ArrayPool<byte>.Shared.Return(_array);
    }

    [InlineArray(StackBufferSize)]
    private struct Buffer
    {
        [SuppressMessage("CodeQuality", "IDE0051:Remove unused private members")]
        private byte _element0;
    }
}
```